appscale/tools/agents/azure_agent.py

```diff
@@ -265,31 +266,77 @@ def describe_instances(self, parameters, pending=False):
     private_ips = []
     instance_ids = []
 
-    public_ip_addresses = network_client.public_ip_addresses.list(resource_group)
-    for public_ip in public_ip_addresses:
-      public_ips.append(public_ip.ip_address)
+    public_ip_addresses, error = self.run_and_return_exception(
+        network_client.public_ip_addresses.list, resource_group)
+    self.raise_if_error(error, "Error calling describe instances")
+
+    public_ips_results = self.convert_to_list(
+        public_ip_addresses, raise_exception=True,
+        err_msg="Error calling describe instances")
+
+    public_ips = [x.ip_address for x in public_ips_results]
+
+    network_interfaces, error = self.run_and_return_exception(
+        network_client.network_interfaces.list, resource_group)
+    self.raise_if_error(error, "Error calling describe instances")
+
+    network_interfaces_results = self.convert_to_list(
```

Comment:
I like the convert_to_list :), its especially nice for Azure because we have to go through a bunch of loops to get some information at times.

Reply:
😄